oneview_redfish_toolkit/api/network_device_function_collection.py

```diff
@@ -0,0 +1,67 @@
+# -*- coding: utf-8 -*-
+
+# Copyright (2017) Hewlett Packard Enterprise Development LP
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+from oneview_redfish_toolkit.api.redfish_json_validator \
+    import RedfishJsonValidator
+
+
+class NetworkDeviceFunctionCollection(RedfishJsonValidator):
+    """Creates a Network Device Function Collection Redfish dict
+
+        Populates self.redfish with a NetworkDeviceFunctionCollection
+    """
+
+    SCHEMA_NAME = 'NetworkDeviceFunctionCollection'
+
+    def __init__(self, device_id, server_hardware):
+        """NetworkDeviceFunctionCollection constructor
+
+            Populates self.redfish and calls _validate()
+
+            Args:
+                server_hardware: a server hardware dict from OneView
+        """
+        super().__init__(self.SCHEMA_NAME)
+        self.redfish["@odata.type"] = \
+            "#NetworkDeviceFunctionCollection.NetworkDeviceFunctionCollection"
+        self.redfish["Name"] = "Network Device Function Collection"
+        members_count = 0
+        self.redfish["Members"] = list()
+        index = int(device_id) - 1
+        devices = server_hardware["portMap"]["deviceSlots"][index]
+        for port in devices["physicalPorts"]:
+            physical_port = str(port["portNumber"])
+            for virtual_port in port["virtualPorts"]:
```

Comment:
You might check with Ricardo but I can't remember if we were going to look at virtual ports.

Reply:
@danielreed I though our final agreement was that the NetworkDeviceFunction was a one to one mapping with each virtual port... is it something else?